### ISSUE TITLE
[Event Hubs Client] Revisit Tests with Class-Level State and Cleanup

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#5986]
# Summary

A number of tests within the Event Hubs client suite use the pattern of creating class-level messaging entities within each test and relying on the `IDisposible` mechanism to clean up those instances, which makes the flow less clear and, depending on the test runner in use, potentially impacts the ability to run tests in parallel. 

# Scope of Work

- Revisit tests that make use of class-level members to hold instances that are specific to a given test case.  These should be rescoped to the test itself, with the goal of having tests be as self-contained as possible without reliance on shared state.

- Ensure that any class-level members that remain are used in a read-only manner or using another approach that allows for tests to run in parallel without corrupting shared state.

- In the cases where it was possible to completely remove class-level members and shared state, remove the `IDisposable` implementation

# Out of Scope

- Refactoring of the general approach for testing, the test cases, code style, and other artifacts of the codebase; changes should be kept as tightly scoped and as minimal as possible.

- Adding of additional test cases or new testing scenarios.

# Success Criteria

- Tests using class-level members have been reviewed and adjusted.  Any remaining class-level members are encapsulating a shared resource with non-trivial cost to create.  

- Any remaining class-level members are used in a manner that allows tests to run in parallel without corrupting them.

- Any unnecessary implementations of `IDisposable` have been removed.   

- Any remaining `IDisposable` implementations perform the correct work to clean shared resources and do not unnecessarily interact with garbage collection or other ceremony, unless there are unmanaged resources and justification to do so.
